aten/src/ATen/cuda/CUDAApplyUtils.cuh

```diff
@@ -943,5 +944,227 @@ bool CUDA_tensor_apply4(at::Tensor a,
   return true;
 }
 
+#define MIN_NUMBER_BINS_FOR_GLOBAL_MEM 5000
```

Comment:
All code below seems very specific to bincount. Can we move it to SummaryOps.cu?